refactor(struct-validators): adicionar novos modelos e validação

MatheusVp2 opened the pull request:
- Criado o pacote 'model' com os seguintes novos arquivos:
  - key.go: define a estrutura básica do modelo de chave.
  - valt.go: define a estrutura do modelo Valt com campos como ID, Nome, Descrição, etc.
  - user.go: define a estrutura do modelo User com campos como ID, Nome, Email, etc.
  - credential.go: define a estrutura do modelo Credential com campos como ID, Nome, Username, etc.

- Criado o pacote 'validator' com novos arquivos para validação:
  - singletonValidator.go: implementa um padrão singleton para o validador.
  - fiberStructValidator.go: implementa um validador de estrutura para o Fiber.
  - langs/factory.go: define funções para configurar tradutores de idiomas.
  - langs/en.go e langs/ptbr.go: implementam traduções para inglês e português brasileiro.

- Removidos arquivos desnecessários do diretório .idea e do pacote internal/http/res.

Essas alterações introduzem novos modelos que são fundamentais para a estrutura de dados do projeto, além de implementar um sistema de validação robusto que melhora a integridade dos dados. A remoção de arquivos desnecessários também ajuda a manter o repositório mais limpo e organizado, facilitando a manutenção futura.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Novos Recursos**
  - A API agora retorna respostas em formato JSON padronizado, incluindo um endpoint de saúde que retorna status e mensagem de boas-vindas.
  - Mensagens de validação e de erro foram aprimoradas para fornecer feedback mais claro ao usuário.
  - Foi implementado suporte multilíngue para mensagens, com traduções para inglês e português (Brasil).

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MatheusVp2 committed:
refactor(struct-validators): adicionar novos modelos e validação

- Criado o pacote 'model' com os seguintes novos arquivos:
  - key.go: define a estrutura básica do modelo de chave.
  - valt.go: define a estrutura do modelo Valt com campos como ID, Nome, Descrição, etc.
  - user.go: define a estrutura do modelo User com campos como ID, Nome, Email, etc.
  - credential.go: define a estrutura do modelo Credential com campos como ID, Nome, Username, etc.

- Criado o pacote 'validator' com novos arquivos para validação:
  - singletonValidator.go: implementa um padrão singleton para o validador.
  - fiberStructValidator.go: implementa um validador de estrutura para o Fiber.
  - langs/factory.go: define funções para configurar tradutores de idiomas.
  - langs/en.go e langs/ptbr.go: implementam traduções para inglês e português brasileiro.

- Removidos arquivos desnecessários do diretório .idea e do pacote internal/http/res.

Essas alterações introduzem novos modelos que são fundamentais para a estrutura de dados do projeto, além de implementar um sistema de validação robusto que melhora a integridade dos dados. A remoção de arquivos desnecessários também ajuda a manter o repositório mais limpo e organizado, facilitando a manutenção futura.

MatheusVp2 commented:
@coderabbitai full review

MatheusVp2 committed:
refactor(model): alterar tipos de data para time.Time

- Modificado o tipo de campo 'CreatedAt' de string para time.Time em 'credential.go'
- Modificado o tipo de campo 'UpdatedAt' de string para time.Time em 'credential.go'
- Modificado o tipo de campo 'CreatedAt' de string para time.Time em 'user.go'
- Modificado o tipo de campo 'UpdatedAt' de string para time.Time em 'user.go'
- Modificado o tipo de campo 'CreatedAt' de string para time.Time em 'valt.go'
- Modificado o tipo de campo 'UpdatedAt' de string para time.Time em 'valt.go'

Essas alterações melhoram a precisão e a manipulação das datas nos modelos, permitindo um melhor gerenciamento de dados temporais e facilitando operações relacionadas a datas no futuro.

MatheusVp2 commented on internal/handler/auth.go:
```diff
@@ -63,20 +64,22 @@ func setAuthCookie(c fiber.Ctx, token string, duration time.Duration) {
 	var req model.CreateUserRequest
 	if err := c.Bind().Body(&req); err != nil {
-		return handleError(c, fiber.StatusBadRequest, "Invalid request body")
+		return err
```
Nesse caso, vai para um middleware global de validação e mapeamento de erro, deve manter esse return err

MatheusVp2 committed:
feat(model): adicionar campos de data ao modelo Key

- Adicionado o campo CreatedAt do tipo time.Time ao modelo Key.
- Adicionado o campo UpdatedAt do tipo time.Time ao modelo Key.

Essas alterações melhoram o modelo Key, permitindo o rastreamento de quando uma chave foi criada e atualizada, o que é essencial para a gestão e auditoria de dados no sistema.

vinirossado commented on internal/http/response/httpResponse.go:
```diff
@@ -0,0 +1,28 @@
+
+import "github.com/gofiber/fiber/v3"
+
+type HttpResponse struct {
```
Based on Static check it should be HTTPResponse instead of HttpResponse

vinirossado replied:
Also, I recommend that you build and run the project locally by running:
`make code-quality`, because there are a few other issues (build/checkers) left behind.
I won’t be able to mention them here since it only shows the files that have been changed.

vinirossado commented on internal/domain/model/credential.go:
```diff
@@ -0,0 +1,17 @@
+	EncryptPassword string `json:"encrypt_password" bson:"encrypt_password"`
+	URL             string `json:"url" bson:"url"`
+
+	ValtID string `json:"valt_id" bson:"valt_id"`
```
There is a typo here, it's VaultID (Missing "U")

vinirossado replied:
Don't forgot to change the filename to match the "Vault" word

vinirossado commented on pkg/validator/langs/en.go:
```diff
@@ -0,0 +1,24 @@
+
+	"github.com/go-playground/locales/en"
+	ut "github.com/go-playground/universal-translator"
+	en_translations "github.com/go-playground/validator/v10/translations/en"
```
Are we going to use snake_case for this sort of variables?

vinirossado commented on pkg/validator/langs/ptbr.go:
```diff
@@ -0,0 +1,24 @@
+
+	"github.com/go-playground/locales/pt_BR"
+	ut "github.com/go-playground/universal-translator"
+	pt_BR_translations "github.com/go-playground/validator/v10/translations/pt_BR"
```
Are we going to use snake_case for this sort of variables?

MatheusVp2 committed:
feat(revision): adicionar modelo Vault e corrigir nomes de campos

- Adicionado o modelo Vault em internal/domain/model/vault.go.
- Corrigido o nome do campo ValtID para VaultID em internal/domain/model/credential.go.
- Atualizado o tipo de resposta para HTTPResponse em vários manipuladores (auth.go, password.go, default.go, validation.go).
- Removido o arquivo de resposta res/response.go.
- Adicionadas novas dependências no go.mod para suporte a validação de idiomas.

Essas alterações introduzem um novo modelo de dados para gerenciar informações de cofres, melhorando a estrutura do código e a clareza dos nomes de campos. A atualização dos manipuladores para usar a nova estrutura de resposta também melhora a consistência e a legibilidade do código, facilitando a manutenção futura. Além disso, a inclusão de bibliotecas para suporte a idiomas amplia a funcionalidade do sistema.
